### PR TITLE
Clean up consensus

### PR DESCRIPTION
**Pull request by shawntabrizi**
For issue #1021

### CODE REVIEW

File: core/client/src/lib.rs [hunk 1]
```diff
@@ -91,9 +89,9 @@ pub mod light;
 #[cfg(feature = "std")]
 mod leaves;
 #[cfg(feature = "std")]
-mod call_executor;
+pub mod call_executor;
```
Comment (bkchr) [inline core/client/src/lib.rs]:
```suggestion 
mod call_executor;
```
The required trait etc is already exported some lines below.

File: core/client/src/lib.rs [hunk 2]
```diff
@@ -91,9 +89,9 @@ pub mod light;
...
 #[cfg(feature = "std")]
-mod client;
+pub mod client;
```
Comment (bkchr) [inline core/client/src/lib.rs]: Same

Comment (bkchr) [reply core/client/src/lib.rs]: Reexport `BlockOrigin` and `ImportResult` there as well.

File: core/consensus/common/src/block_import.rs [hunk 3]
```diff
@@ -1,39 +1,16 @@
...
-	AlreadyInChain,
-	/// Block or parent is known to be bad.
-	KnownBad,
-	/// Block parent is not in the chain.
-	UnknownParent,
-}
+use client::backend;
+use client::client::{BlockOrigin, ImportResult, PrePostHeader, Client};
+use client::blockchain::{self, HeaderBackend};
+use client::call_executor::CallExecutor;
```
Comment (bkchr) [inline core/consensus/common/src/block_import.rs]:
```suggestion
use client::CallExecutor;
```
Do the same for line 8.

Comment (bkchr) [reply core/consensus/common/src/block_import.rs]: (and merge them)

File: core/client/src/client.rs [hunk 4]
```diff
@@ -585,7 +628,7 @@ impl<B, E, Block, RA> Client<B, E, Block, RA> where
 		})
 	}
 
-	fn execute_and_import_block(
+	pub fn execute_and_import_block(
```
Comment (bkchr) [inline core/client/src/client.rs]: I would propose to move this function as well into `consensus`.

File: core/client/src/client.rs [hunk 5]
```diff
@@ -126,6 +126,38 @@ pub enum BlockStatus {
 	Unknown,
 }
 
+/// Block import result.
+#[derive(Debug)]
```
Comment (bkchr) [inline core/client/src/client.rs]: If you move `execute_and_import_block` into `consensus`, please move this as well.

File: core/consensus/common/src/block_import.rs (hunk 3 above)
Comment (shawntabrizi) [reply core/consensus/common/src/block_import.rs]: These fixes should be done :) 👍

**Comment by bkchr** [comment]
@shawntabrizi Looks good, but you still need to fix the errors and also add missing documentation.

**Comment by shawntabrizi** [comment]
Will do once all the PoC-3 changes are done so I am not chasing new errors afterward

**Comment by gavofyork** [comment]
@rphmeier / @gnunicorn ^^^

**Comment by rphmeier** [comment]
Chatted with @shawntabrizi about this on Riot -- basically, we shouldn't have `execute_and_import_block` as part of `BlockImport` because that's what `import_block` is supposed to do already, and indeed `execute_and_import_block` is used for the implementation of `import_block` in `Client`.

As I mentioned in the issue, I think it does make sense to keep `client` dependent on `consensus-common`.

**Comment by shawntabrizi** [comment]
I will close this for now, and chat with Ben again after the holidays to see if these changes are still needed.

**Pull request closed without merging**